spikeinterface/preprocessing/highpass_spatial_filter.py

```diff
@@ -0,0 +1,290 @@
+                    "The agc_options dict must contain both the 'window_length_s' field"
+                agc_options["sampling_interval"] = sampling_interval
+
+        # Pre-compute spatial filtering parameters
+        butter_kwargs_default = {'btype': 'highpass', 'N': 3, 'Wn': 0.01}
+        if butter_kwargs is not None:
+            assert all(k in ("N", "Wn") for k in butter_kwargs), \
+                "butter_kwargs can only specify filter order 'N' and critical frequency 'Wn' (1 is Nyquist)"
+            butter_kwargs_default.update(butter_kwargs)
+        sos_filter = scipy.signal.butter(**butter_kwargs_default, output='sos')
+
+        for parent_segment in recording._recording_segments:
+            rec_segment = HighPassSpatialFilterSegment(parent_segment,
+                                                       n_channel_pad,
+                                                       n_channel_taper,
+                                                       agc_options,
+                                                       sos_filter,
+                                                       order_f,
+                                                       order_r)
+            self.add_recording_segment(rec_segment)
+
+        self._kwargs = dict(recording=recording.to_dict(),
+                            n_channel_pad=n_channel_pad,
+                            n_channel_taper=n_channel_taper,
+                            direction=direction,
+                            agc_options=agc_options,
+                            butter_kwargs=butter_kwargs)
+
+class HighPassSpatialFilterSegment(BasePreprocessorSegment):
+
+    def __init__(self,
+                 parent_recording_segment,
+                 n_channel_pad,
+                 n_channel_taper,
+                 agc_options,
+                 sos_filter,
+                 order_f,
+                 order_r
+                 ):
+        self.parent_recording_segment = parent_recording_segment
+        self.n_channel_pad = n_channel_pad
+        self.n_channel_taper = n_channel_taper
+        self.agc_options = agc_options
+        self.order_f = order_f
+        self.order_r = order_r
+        # get filter params
+        self.sos_filter = sos_filter
+
+    def get_traces(self, start_frame, end_frame, channel_indices):
+        if channel_indices is None:
+            channel_indices = slice(None)
+        traces = self.parent_recording_segment.get_traces(start_frame,
+                                                          end_frame,
+                                                          slice(None))
+        if self.order_f is not None:
+            traces = traces[:, self.order_f]
+        else:
+            traces = traces.copy()
+
+        traces = kfilt(traces,
+                       self.n_channel_pad,
+                       self.n_channel_taper,
+                       self.agc_options,
+                       self.sos_filter)
```

Comment:
since this function is mainly a padding and a sosfiltfilt I would not make a call to a function here.
I would move the logic of kfilt just here.
So we can have a better control on what is computed once (in the init) and at evry chunk.